businessRules/donations.js

```diff
@@ -0,0 +1,53 @@
+const { v4 } = require('uuid');
+const { createDonation } = require('../models/donations');
+const { updateIdea, getAllIdeas } = require('../models/ideas');
+
+function createDonationsWithDataValidation(firstName, sum, userId, ideaId) {
+    const generateId = v4();
+    const anonymous = 'Anonymous';
+
+    if (firstName && userId) {
+        createDonation(firstName, sum, userId, ideaId);
+    }
+
+    if (!firstName && !userId) {
+        firstName = anonymous;
+        userId = generateId;
+        createDonation(firstName, sum, userId, ideaId);
+    }
+
+    if (!userId) {
+        userId = generateId;
+        createDonation(firstName, sum, userId, ideaId);
+    }
+
+    if (!firstName) {
+        firstName = anonymous;
+        createDonation(firstName, sum, userId, ideaId);
```

Comment:
Try to refactor and call `createDonation` only once.

Reply:
Updated and tested. )